### ISSUE TITLE
VS 16.5.0 Preview 5.0 - IDE0071 triggered on non-constant value

### ISSUE THREAD

**Issue by GGG-KILLER** [dotnet/roslyn#42247]
**Version Used**: VS Community 16.5.0 Preview 5.0

**Steps to Reproduce**:
1. Create a new .NET Core 3.1 console project
2. Put the following code anywhere:
```cs
using System;
using System.Linq;

public static class Sample
{
    public static void PrintRightAligned ( String[] strings )
    {
        var maxLength = strings.Max(str => str.Length);

        for ( var i = 0; i < strings.Length; i++ )
        {
            var str = strings[i];
            Console.WriteLine ( $"{i}. {str.PadRight ( maxLength, ' ' )}" );
        }
    }
}
```

**Expected Behavior**:
IDE0071 is not triggered since `maxLength` is not a constant.

**Actual Behavior**:
IDE0071 is triggered:
![image](https://user-images.githubusercontent.com/5892127/76152369-adc93c80-609d-11ea-9d8c-61ba68835ada.png)

And the code fix results in error-generating code:
![image](https://user-images.githubusercontent.com/5892127/76152364-9db15d00-609d-11ea-95bc-e9774d43090b.png)

**Comment by CyrusNajmabadi** [comment]
On it.

**Comment by CyrusNajmabadi** [comment]
Thanks for the great repro :)

**Comment by GGG-KILLER** [comment]
Glad to help! (though I wish I could've helped further than just giving a repro)
Thanks for fixing this so quickly
